distribution/docker/src/docker/Dockerfile

```diff
@@ -61,10 +61,14 @@ RUN apk add gnupg gcc make musl-dev openssl-dev openssl-libs-static file
 RUN mkdir /work
 WORKDIR /work
 
-# Fetch curl sources and files for validation
-RUN wget "https://daniel.haxx.se/mykey.asc" -O "curl-gpg.pub" && \\
-    wget "\${TARBALL_URL}.asc" -O "\${TARBALL_PATH}.asc" && \\
-    wget "\${TARBALL_URL}" -O "\${TARBALL_PATH}"
+# Fetch curl sources and files for validation. Note that alpine's `wget` doesn't have retry options.
+RUN for iter in {1..10}; do \\
+      wget "https://daniel.haxx.se/mykey.asc" -O "curl-gpg.pub" && \\
+      wget "\${TARBALL_URL}.asc" -O "\${TARBALL_PATH}.asc" && \\
+      wget "\${TARBALL_URL}" -O "\${TARBALL_PATH}" && \\
```

Comment:
A small note: one issue here is that if any of the three wgets fails here, the entire loop will fail and all downloads will be retried. Is this deliberate? Alternatively we'd have to split the wget's within separate loops.

Reply:
These are idempotent operations so that's not a huge deal but you do risk that one fails one iteration and another fails the next. This is still an improvement though. Wrapping each of these is going to start getting a bit silly unless we can somehow encapsulate this logic in a function of some kind.

Reply:
Yeah and while having an inline function within the scope of RUN is possible it'll look convoluted. I agree this is already an improvement.

Reply:
@dliappis Actually the inline function isn't too bad - I've pushed a change, see what you think.

Reply:
@pugnascotia Looks MUCH better, thank you!